src/tiledb/cloud/vcf/ingestion.py

```diff
@@ -1398,14 +1456,20 @@ def ingest(
     """
 
     # Validate user input
-    if not search_uri and not sample_list_uri:
-        raise ValueError("Either `search_uri` or `sample_list_uri` must be provided.")
+    if not search_uri and not sample_list_uri and not metadata_uri:
+        raise ValueError(
+            "One of `search_uri`, `sample_list_uri`, or `metadata_uri`"
+            " must be provided."
+        )
 
-    if search_uri and sample_list_uri:
-        raise ValueError("Cannot specify both `search_uri` and `sample_list_uri`.")
+    if sum([bool(search_uri), bool(sample_list_uri), bool(metadata_uri)]) > 1:
```

Comment:
How about `== 1` and say “Exactly one of ...”?

Reply:
Good idea, I can collapse 2 checks into one.